daemon/src/maker_cfd.rs

```diff
@@ -345,6 +355,21 @@ where
             })
             .await?;
 
+        if !self.connected_takers.insert(taker_id) {
+            anyhow::bail!("Taker already connected: {:?}", &taker_id);
```

Comment:
Is this really a failure as such?

Reply:
It's not a failure, but an indication that something is not right with our code....

I've downgraded both to `tracing::warn!`, ideally I'd probably add a `debug_assert!` there to catch such inconsistencies early if they'd happen.